migrations/2025-04-25-090000_trusted-publishing/up.sql

```diff
@@ -0,0 +1,64 @@
+create table github_oidc_configs
+(
+    id serial primary key,
+    created_at timestamptz not null default now(),
+    crate_id int not null references crates on delete cascade,
+    repository_owner varchar not null,
+    repository_owner_id int not null,
+    repository_name varchar not null,
+    workflow_filename varchar not null,
+    environment varchar
+);
+
+comment on table github_oidc_configs is 'Trusted Publisher OIDC configuration for GitHub Actions';
+comment on column github_oidc_configs.id is 'Unique identifier of the `github_oidc_configs` row';
+comment on column github_oidc_configs.created_at is 'Date and time when the configuration was created';
+comment on column github_oidc_configs.crate_id is 'Unique identifier of the crate that this configuration is for';
+comment on column github_oidc_configs.repository_owner is 'GitHub name of the user or organization that owns the repository';
+comment on column github_oidc_configs.repository_owner_id is 'GitHub ID of the user or organization that owns the repository';
+comment on column github_oidc_configs.repository_name is 'Name of the repository that this configuration is for';
+comment on column github_oidc_configs.workflow_filename is 'Name of the workflow file inside the repository that will be used to publish the crate';
+comment on column github_oidc_configs.environment is 'GitHub Actions environment that will be used to publish the crate (if `NULL` the environment is unrestricted)';
+
+create index github_oidc_configs_repository_owner_id_repository_name_index
+    on github_oidc_configs (repository_owner_id, repository_name);
```

Comment:
I think it would be better to delay creating the index until we have our endpoint and SQL query, as we might change it to better suit our queries (e.g., if we want it to be a covering index).

Reply:
you're probably right. I've removed the non-unique index for now.